code/modules/cargo/bounty.dm

```diff
@@ -30,50 +30,53 @@
  */
 /proc/random_bounty(guided = 0)
 	var/bounty_num
-	if(guided && (guided != CIV_JOB_RANDOM))
-		bounty_num = guided
-	else
-		bounty_num = rand(1,12)
-	switch(bounty_num)
-		if(1)
-			var/subtype = pick(subtypesof(/datum/bounty/item/assistant))
-			return new subtype
-		if(2)
-			var/subtype = pick(subtypesof(/datum/bounty/item/mech))
-			return new subtype
-		if(3)
-			var/subtype = pick(subtypesof(/datum/bounty/item/chef))
-			return new subtype
-		if(4)
-			var/subtype = pick(subtypesof(/datum/bounty/item/security))
-			return new subtype
-		if(5)
-			if(rand(2) == 1)
-				return new /datum/bounty/reagent/simple_drink
-			return new /datum/bounty/reagent/complex_drink
-		if(6)
-			if(rand(2) == 1)
-				return new /datum/bounty/reagent/chemical_simple
-			return new /datum/bounty/reagent/chemical_complex
-		if(7)
-			var/subtype = pick(subtypesof(/datum/bounty/virus))
-			return new subtype
-		if(8)
-			if(rand(2) == 1)
-				var/subtype = pick(subtypesof(/datum/bounty/item/science))
-				return new subtype
-			var/subtype = pick(subtypesof(/datum/bounty/item/slime))
-			return new subtype
-		if(9)
-			var/subtype = pick(subtypesof(/datum/bounty/item/engineering))
-			return new subtype
-		if(10)
-			var/subtype = pick(subtypesof(/datum/bounty/item/mining))
-			return new subtype
-		if(11)
-			var/subtype = pick(subtypesof(/datum/bounty/item/medical))
-			return new subtype
-		if(12)
-			var/subtype = pick(subtypesof(/datum/bounty/item/botany))
-			return new subtype
-
+	var/chosen_type
+	var/bounty_succeeded = FALSE
+	var/datum/bounty/item/bounty_ref
+	while(!bounty_succeeded)
+		if(guided && (guided != CIV_JOB_RANDOM))
+			bounty_num = guided
+		else
+			bounty_num = rand(1,13)
```

Comment:
This is my mistake from the past coming to bite me, but you should probably make the 13 into a define so that it's not a magic number, and comment it to let people know to increase it if they're adding a new category of bounty. Open to suggestions however.

Reply:
converted all of it to defines